ledger-core-ethereum/inc/ethereum/ERC20/ERC20LikeOperation.hpp

```diff
@@ -139,27 +150,26 @@ namespace ledger {
             };
 
             ERC20LikeOperation &setBlockHeight(int64_t blockHeight) {
-                _blockHeight = blockHeight;
+                if (!block.hasValue()) {
+                    block = api::Block();
+                }
+
+                block->height = blockHeight;
+
                 return *this;
             }
-            
+
+            void refreshUid(std::string const&) override;
+
         private:
-            std::string _uid;
             std::string _ethUidOperation;
             std::string _hash;
             std::shared_ptr<api::BigInt> _nonce;
             std::shared_ptr<api::BigInt> _gasPrice;
             std::shared_ptr<api::BigInt> _gasLimit;
             std::shared_ptr<api::BigInt> _gasUsed;
-            std::string _sender;
-            std::string _receiver;
-            std::shared_ptr<api::BigInt> _value;
             std::vector<uint8_t> _data;
-            std::chrono::system_clock::time_point _time;
-            api::OperationType _operationType;
             int32_t _status;
-            int64_t _blockHeight;
```

Comment:
Why all these members are deleted ?

Reply:
As discussed out loud, those come from `Operation` (the common type), since we have to inherit from it! :)